### PR TITLE
add parameter type declarations to private methods

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Bridge/Twig/Command/DebugCommand.php [hunk 1]
```diff
@@ -248,7 +248,7 @@ private function displayGeneralText(SymfonyStyle $io, string $filter = null)
         }
     }
 
-    private function displayGeneralJson(SymfonyStyle $io, $filter)
+    private function displayGeneralJson(SymfonyStyle $io, string $filter = null)
```
Comment (derrabus) [inline src/Symfony/Bridge/Twig/Command/DebugCommand.php]:
```suggestion
    private function displayGeneralJson(SymfonyStyle $io, ?string $filter)
```

Why do you introduce a default value?

**Review by derrabus — CHANGES_REQUESTED** [review]

**Comment by xabbuh** [comment]
Status: Needs review

**Review by nicolas-grekas — CHANGES_REQUESTED** [review]
deps=low line on Travis consistently fails with a timeout, this looks related, isn't it?

**Review by nicolas-grekas — APPROVED** [review]
(we investigated that the failure is a pure CI artefact and should be fixed after merge)

**Comment by Tobion** [comment]
I think it's good to do that here. But also I think it's a huge risk to add incompatibilities that are not covered by tests. I would not trust the phpdocs all the time and private mostly had no phpdocs at all. So the only way the add types there is by tracing all the entry points individually which I guess was a huge pain and is almost impossible to review.
For changes like this, I wish we had phpstan active. That would give us much more confidence.

**Comment by fabpot** [comment]
As this is for 4.4, we will have plenty of people testing this in real life, which is way much better than any static analysis tool.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

File: src/Symfony/Component/HttpFoundation/Session/Storage/Handler/PdoSessionHandler.php [hunk 2]
```diff
@@ -775,13 +771,9 @@ private function getSelectSql(): string
     /**
      * Returns an insert statement supported by the database for writing session data.
      *
-     * @param string $sessionId   Session ID
-     * @param string $sessionData Encoded session data
-     * @param int    $maxlifetime session.gc_maxlifetime
```
Comment (ostrolucky) [inline src/Symfony/Component/HttpFoundation/Session/Storage/Handler/PdoSessionHandler.php]: quite a lot of useful phpdoc descriptions have been removed :/

Comment (derrabus) [reply src/Symfony/Component/HttpFoundation/Session/Storage/Handler/PdoSessionHandler.php]: Which of those would you like to keep?

Comment (ostrolucky) [reply src/Symfony/Component/HttpFoundation/Session/Storage/Handler/PdoSessionHandler.php]: maxlifetime

**Pull request merged**